### DISCUSSION TITLE
When access adminweb EJBCA "No client certificate was presented"

### DISCUSSION THREAD

**Discussion by alandrma** [Keyfactor/ejbca-ce#386] [General]
Hi everyone,
I want to ask regarding accessing the ejbca web admin.

i have deployed ejbca v8 with wildfly 26.
using mariadb with connector version 2.7.0

When the process of deploying and importing the superadmin certificate that has been generated by EJBCA has been completed into Firefox, it still doesn't work. there is an error as follows.

<img width="705" alt="image" src="https://github.com/Keyfactor/ejbca-ce/assets/40722633/117b161f-1a37-4540-a696-9e83cec3b3b8">

and when I checked the certificate in Firefox when accessing adminweb, Firefox didn't detect the superadmin that had been imported previously. In fact, the certificate in Firefox is detected as localhost cert, not superadmin which was previously generated. Please help, do you think there are any wrong steps? Thank You

Edited : I was using RSA 2048 with management CA

**Comment by primetomas** [comment]
If you can't see the client certificate, looking in the right place, in FireFox it wasn't installed.
Are you accessing wildfly on port 8443? If the superadmin.p12 (not only a certificate, but the p12 with the private key as well) was imported you should be prompted to select the certificate.

Firefox->Settings->Privacy&security->View certificates->Your certificates.

**Reply by alandrma** [reply]
The client certificate has been imported and has been successful without any problems. I also have another EJBCA client server and it is fine.

However, when accessing the superadmin web admin, a popup to select superadmin cert does not appear. instead making the cert as localhost which is not suitable for accessing the adminweb

i also checked in this discussion forum, still haven't found the solving process.

Is there a problem with TLS that needs to be reconfigured in the subsystem?

**Reply by primetomas** [reply]
localhost would be your server certificate for the admin web. This depends on how you configured things during installation. But in worst case this should only give you a warning in the web browser that the name is not the same as the URL you are visiting. the superadmin.p12 you imported you never be named superadmin, unless you have configured that as superadmin DN in install.properties.

**Reply by primetomas** [reply]
Could be the TLS subsystem configuration. You can always troubleshoot TLS connections, seing what CA certificates they accept and such, using "openssl s_client". Other potential issues (if you configured you server in that way) would be that you have a truststore on the wildfly server that does not include you ManagementCA (whatever CA you issued your superadmin.p12 with).

There are many ways to do things wrong with TLS, and a bit hard to debug due to the encrypted nature of things. But usually there is a issmatch between client and server CAs somewhere.

I'm assuming that your screenshot is from something like "https://mycahost:8443/ejbca/adminweb"

**Reply by alandrma** [reply]
@primetomas Hello, 

referring here [Automatic Self-Signed Certificate Generation with Elytron
](https://wildfly-security.github.io/wildfly-elytron/blog/auto-self-signed-certificate-generation/) it seems I haven't configured the default SSL so what is detected when accessing the ejbca adminweb reads local host, following the link I provided above, how do I configure this to change the default selfsign certificate to a certificate signed by CA management on the wildfly 26 subsystem?

thank you very much for the help

**Reply by alandrma** [reply]
for the log self sign certificate 
<img width="1369" alt="image" src="https://github.com/Keyfactor/ejbca-ce/assets/40722633/7a46acc5-9d01-4bf1-be70-5e4e9bb336dc">

**Reply by primetomas** [reply]
Yeah, you need to follow the installation instructions all the way through the end.
https://doc.primekey.com/ejbca/ejbca-installation, which includes TLS setup with the server certificate generated.

**Reply by gilles380** [reply]
hi @primetomas 
I need some clarification on how I can modify install.properties AFTER I have my ejbca running. 
As told in the documentation, it is not needed to recompile right ?  (just restart WildFly server ?)
Because, I tried to access with REST API with Curl **from an external host** and it come with :
curl: (60) SSL: no alternative certificate subject name matches target host name '192.168.x.y'
Also , external web access to ejbca admin GUI does not work either,
What would be your recommendation to have external access working (I dot not have any apache proxy, just my local network as it is)
thanks for your support

**Reply by primetomas** [reply]
install.properties is completely useless after you have done the first installation.

What you can do is that you can re-issue the TLS server certificate by editing the end entity 'tomcat' and setting other values for dnsName.
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ca-operations-guide/end-entities/ssl-certificate-expiration

**Reply by merryo11** [reply]
What if we change the local SSL under management CA with commercial CA SSL like under digicert, sectigo or let's encrypt

**Reply by primetomas** [reply]
Then you will simply use those keystores in your WildFly/JBoss installation instead of ones generated by EJBCA. These are simply jks/p12 keystore files copied to a WildFly directory. You can use any keys/certificates you like for TLS in WildFly.
